packages/astro/src/prerender/routing.ts

```diff
@@ -0,0 +1,67 @@
+import type { AstroSettings, RouteData } from '../@types/astro';
+import { preload, type DevelopmentEnvironment } from '../core/render/dev/index.js';
+import { getPrerenderStatus } from './metadata.js';
+
+type GetSortedPreloadedMatchesParams = {
+	env: DevelopmentEnvironment;
+	matches: RouteData[];
+	settings: AstroSettings;
+};
+export async function getSortedPreloadedMatches({
+	env,
+	matches,
+	settings,
+}: GetSortedPreloadedMatchesParams) {
+	return (
+		await preloadAndSetPrerenderStatus({
+			env,
+			matches,
+			settings,
+		})
+	).sort((a, b) => prioritizePrerenderedMatchesComparator(a.route, b.route));
+}
+
+type PreloadAndSetPrerenderStatusParams = {
+	env: DevelopmentEnvironment;
+	matches: RouteData[];
+	settings: AstroSettings;
+};
+async function preloadAndSetPrerenderStatus({
+	env,
+	matches,
+	settings,
+}: PreloadAndSetPrerenderStatusParams) {
+	const preloaded = await Promise.all(
+		matches.map(async (route) => {
```

Comment:
Putting a note: We had issues in the past that using `Promise.all` on promises that access the filesystem could slow things down and even error out. But IIUC `matches` here should usually be a low number, so this should be fine.

Reply:
Yeah `matches` should generally be low, I'll keep an eye out if this every becomes problematic 👍️